src/GeometryInstancing/Program.cs

```diff
@@ -0,0 +1,261 @@
+            _graphicsDevice.UpdateBuffer(_cameraProjViewBuffer,64,_camera.ProjectionMatrix);
+
+            VertexLayoutDescription vertexLayout 
+                = new VertexLayoutDescription(
+                    new VertexElementDescription("Position",VertexElementSemantic.Position,VertexElementFormat.Float2),
+                    new VertexElementDescription("Colour",VertexElementSemantic.Color,VertexElementFormat.Float4)
+                );
+            
+            VertexElementDescription vertexElementPerInstance
+                = new VertexElementDescription("xOff",VertexElementSemantic.Position,VertexElementFormat.Float1);
+
+            VertexLayoutDescription vertexLayoutPerInstance 
+                = new VertexLayoutDescription(
+                    stride: 4,
+                    instanceStepRate: 1,
+                    elements: new VertexElementDescription[] {vertexElementPerInstance}
+                );
+
+            _vertexShader = LoadShader(ShaderStages.Vertex);
+            _fragmentShader = LoadShader(ShaderStages.Fragment);
+
+            GraphicsPipelineDescription pipelineDescription = new GraphicsPipelineDescription(){
+                BlendState = BlendStateDescription.SingleOverrideBlend,
+                DepthStencilState = new DepthStencilStateDescription(
+                    depthTestEnabled: true,
+                    depthWriteEnabled: true,
+                    comparisonKind: ComparisonKind.LessEqual),
+                RasterizerState = new RasterizerStateDescription(
+                    cullMode: FaceCullMode.Back,
+                    fillMode: PolygonFillMode.Solid,
+                    frontFace: FrontFace.Clockwise,
+                    depthClipEnabled: true,
+                    scissorTestEnabled: false
+                ),
+                PrimitiveTopology = PrimitiveTopology.TriangleStrip,
+                ResourceLayouts = new ResourceLayout[] {_resourceLayout},
+                ShaderSet = new ShaderSetDescription(
+                    // The ordering of layouts directly impacts shader layout schemes
+                    vertexLayouts: new VertexLayoutDescription[] {vertexLayout,vertexLayoutPerInstance},
+                    shaders: new Shader[] {_vertexShader,_fragmentShader}
+                ),
+                Outputs = _graphicsDevice.SwapchainFramebuffer.OutputDescription
+            };
+
+            _pipeline = _factory.CreateGraphicsPipeline(pipelineDescription);
+
+            _commandList = _factory.CreateCommandList();
+
+        }
+
+        private static Shader LoadShader(ShaderStages stage)
+        {
+            string extension = null;
+            switch (_graphicsDevice.BackendType)
+            {
+                case GraphicsBackend.Direct3D11:
+                    extension = "hlsl.bytes";
+                    // not implemented
+                    throw new System.InvalidOperationException();

```

Comment:
Feel free to throw a `NotImplementedException`. I can follow up and implement them pretty quickly.